dockstore/dockstore
# Fix slow endpoints with checker workflows

**Description**
Key portion was to change parentEntries in BioWorkflow to be FetchType.Lazy. That led to a need to initialize parentEntries for serialization to JSON to work. We are inconsistent on what collection we initialize, so I tried to follow the existing pattern.

Why is this now slow?
https://jpa-buddy.com/blog/hibernate6-whats-new-and-why-its-important/, which includes "So, in Hibernate 6, prefer the LAZY fetch type to avoid combination explosion for the recursive associations during queries."

Note with the fix #5536, the bug as originally discovered isn't happening for non-checker workflows; it now only happens with checker workflows.

**Review Instructions**
1. Open developer tools in your browser
2. Go to https://qa.dockstore.org/workflows/github.com/DataBiosphere/topmed-workflows/checker_UM_variant_caller_wdl:1.32.0?tab=info
3. Make sure endpoints are all responsive, ideally 1 second max. See screenshot for what it looks like now, with 19-20 second (!) responses and lots of turtles.

![Screen Shot 2023-06-07 at 5 18 24 PM](https://github.com/dockstore/dockstore/assets/1049340/c58dcaaf-b119-46a0-925d-41b21d0d41d6)

**Issue**
SEAB-5552

**Security**
If there are any concerns that require extra attention from the security team, highlight them here.

Please make sure that you've checked the following before submitting your pull request. Thanks!
